tests/unit/acquisition/test_function.py

```diff
@@ -171,6 +173,37 @@ def test_expected_improvement(
     npt.assert_allclose(ei, ei_approx, rtol=rtol, atol=atol)
 
 
+def test_min_value_entropy_search_builder_raises_for_empty_data() -> None:
+    data = Dataset(tf.zeros([0, 1]), tf.ones([0, 1]))
+    search_space = Box([0, 0], [1, 1])
+    with pytest.raises(ValueError):
+        MinValueEntropySearch(search_space).prepare_acquisition_function(
+            data, QuadraticMeanAndRBFKernel()
+        )
```

Comment:
if you do it like this, you can be sure the `ValueError` is raised by the call to `prepare_acquisition_function` not by the call to `__init__`
```suggestion
    builder = MinValueEntropySearch(search_space)
    with pytest.raises(ValueError):
        builder.prepare_acquisition_function(
            data, QuadraticMeanAndRBFKernel()
        )
```